app/src/main/java/chat/rocket/android/push/interactors/DefaultPushInteractor.java

```diff
@@ -1,19 +0,0 @@
-package chat.rocket.android.push.interactors;
-
-import chat.rocket.android.helper.ServerPolicyHelper;
-import chat.rocket.android.model.ServerConfig;
-import chat.rocket.android.realm_helper.RealmStore;
-
-public class DefaultPushInteractor implements PushInteractor {
```

Comment:
And it's not.

It would be better if we have a `Repository` interface and an implementation of that repository using `Realm`.

Another improvement would be making our domain entities not extend `RealmObject` at all.

We could list a lot of improvements but we need to do things a step at a time.

If you want to take the next step, please do it. :)